broadinstitute/orsp-pub
# BTRX - 733 - Convert Sample Data Cohorts Tab

## Addresses
[BTRX-733](https://broadinstitute.atlassian.net/browse/BTRX-733): Convert Sample/Data Cohort Tab
## Changes
- Remove deprecated grails view
- Used [rc-collapse](https://github.com/react-component/collapse) library for collapse functionallity. 
- Created a generic collapsible component where we can put custom components as header and body
- Refactored consent groups list list action
- First use of react redirections. This threw a memory leak warning for trying to update state for unmounted components caused by spinner-service. This is fixed by unsubscribing a specific spinner created by the component using it. 
- Corrected table buttons text to match as they were in old implementations.
**Note:** We still need to check all the places where spinnerService is used, I only fixed the places where ProjectReview / Sample Data Cohorts / New Consent Group / Existing Consent group redirections are, I'm pretty sure there are more, but escapes to this PR scope and size so it would be nice to catch in a sepparate story.
**Note2:**We are still depending on grails redirections somehow, a lot of [these](https://github.com/broadinstitute/orsp-pub/compare/BTRX-733-convertSampleDataCohortsTab#diff-8ed4190862442832525e51e9360e0ab4R42) main_layout redirections will disapear as we advance in react migration.
## Testing
- Create a new project and navigate to Sample/Data Cohorts tab.
- Click on `Add New Sample/ Data Cohorts` button. (this redirection is made by react),
- Refresh page using F5 or browser's button. This shouldn't throw any warning in console.
- Click on browser's back button (return to previous page), this shouldn't throw any warning in console. (Note: Consent group has another async setState not beign handled here and not related to spinnerService)
- Return to `Add New Sample/ Data Cohorts` and create a new one, be sure to add some documents in order to later check them in Project's Review page
- After previous creation, a green message should appear indicating its creation
- The new element should appear in the tab. Check their buttons, Approve, Reject, Unlink, Request Clarification.
- Press the drop down button, the documents that you uploaded should appear here. The name of the file should be clickable, downloading the file.
- Check that Delete button is working
- Only Admin users can Approve, Unlink, Reject and Delete attachments
- Since Spinner Service has been modified, the places to be checked if the spinner still working are:
  . Access Project Review, and Consent group (loading indicator appears)
  . Project Review Admin Only Tab (update some date here and save)
  . Project Review, and Consent group Comments Tab ( add a new comment)
  . Project Review, and Consent group Documents Tab ( upload a new file)
---
